### ISSUE TITLE
no-case-declarations returns duplicate results

### ISSUE THREAD

**Issue by Arcanemagus** [eslint/eslint#10048]
**Tell us about your environment**

* **ESLint Version:** v4.18.2
* **Node Version:** v9.7.1
* **npm Version:** 5.7.1

**What parser (default, Babel-ESLint, etc.) are you using?**

`default`

**Please show your full configuration:**

Configuration:
```json
{
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "no-case-declarations": "error"
  }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export default function myFunction() {
  switch (true) {
    case true:
      const x = 1
      const y = 2
      return { x, y }
    default:
      return null
  }
}
```

```bash
npx eslint .\foo.js
```

You can download the exact files from [this Gist](https://gist.github.com/Arcanemagus/8e7877a6d085f0e9cf9fb286ffc2435d).

**What did you expect to happen?**

A single error message is returned.

**What actually happened? Please include the actual, raw output from ESLint.**
CLI output
```
C:\temp\github\linter-ui-default_GH465\node_modules\eslint\bin\eslint.js

C:\temp\github\linter-ui-default_GH465\foo.js
  3:5  error  Unexpected lexical declaration in case block  no-case-declarations
  3:5  error  Unexpected lexical declaration in case block  no-case-declarations

✖ 2 problems (2 errors, 0 warnings)
```

Note that this isn't just a bug in the default CLI output, as the raw `report` from the `CLIEngine` shows the entire result object is identical:
```
> JSON.stringify(report.results.messages[0]) === JSON.stringify(report.results.messages[1])
true
```

Originally reported as https://github.com/steelbrain/linter-ui-default/issues/465, apologies if this has already been reported as I didn't find it in a search.

**Comment by platinumazure** [comment]
I think the rule is trying to report each of the two declarations, but each report location is on the SwitchCase rather than the variable declaration node. I think it would make sense to report from the variable declarations (lines 4 and 5 in your example).
